### ISSUE TITLE
AWX v7.0.0 local build images not working

### ISSUE THREAD

**Issue by ravigawade** [ansible/awx#8099]
##### ISSUE TYPE
 - Bug report

##### SUMMARY
Have been using AWX for 2+ yrs with official images and v7.0.0 for last 10months in which it uses Ansible 2.8.4 and python 2.7 within the containers venv and everything workd perfectly.  This time I build the images locally using my company's base images centos7. After few hiccups was able to build images successfully, upon deploying they AWX doesnt works as expected. Below are the details:
      - python 2.7 in not installed in the containers venv lib dir but have python 3.6
      - When tried to run the job (playbook) in AWX got error "ansible module not found"
      -  Resolved this error by manually installing ansible=2.8.4 within the containers, which then used python 3.6 as its 
         interpreter by default
      - With python 3.6 AWS modules doesn't work such as Inventory Jobs, create ec2 in playbooks etc
In the official AWX task and web images of v7.0.0 ansible 2.8.4 with python2.7 interpreter is configured by default.  

##### ENVIRONMENT
* AWX version: 7.0.0
* AWX install method: docker for mac
* Ansible version:  2.9.4
* Operating System: Mac OS
* Web Browser: Chrome

##### STEPS TO REPRODUCE
mkdir awx
cd awx
git init
git remote add origin https://github.com/ansible/awx.git
git fetch 
git check out tag/7.0.0
cd installer
ansible-playbook -i inventory install.yml
docker ps

##### EXPECTED RESULTS
1. AWS inventory update job run should successfully get the hosts
2. Successful Job run of  playbook to fetch latest ami and create ec2

This console o/p is from tower v11.0.0 deployed using offical AWX images directly:

1. Where we can see ansible 2.84 using python 2.7 interpreter for the Job run:

` ansible-playbook 2.8.4
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/var/lib/awx/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible-playbook
  python version = 2.7.5 (default, Jun 20 2019, 20:27:34) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]
Using /etc/ansible/ansible.cfg as config file `

2. Below is the AWS inventory update Job run console o/p:
 `   1.715 INFO     Updating inventory 30: my_qa_inventory
    2.054 INFO     Reading Ansible inventory source: /var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/plugins/inventory/ec2.py
    2.056 INFO     Using VIRTUAL_ENV: /var/lib/awx/venv/ansible
    2.056 INFO     Using PATH: /var/lib/awx/venv/ansible/bin:/var/lib/awx/venv/awx/bin:/var/lib/awx/venv/awx/bin:/var/lib/awx/venv/awx/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
    2.056 INFO     Using PYTHONPATH: /var/lib/awx/venv/ansible/lib/python2.7/site-packages:
    3.391 ERROR    ansible-inventory 2.8.4
    3.391 ERROR      config file = /etc/ansible/ansible.cfg
    3.392 ERROR      configured module search path = [u'/var/lib/awx/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
    3.392 ERROR      ansible python module location = /usr/lib/python2.7/site-packages/ansible
    3.392 ERROR      executable location = /usr/bin/ansible-inventory
    3.392 ERROR      python version = 2.7.5 (default, Jun 20 2019, 20:27:34) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]
    3.392 ERROR    Using /etc/ansible/ansible.cfg as config file
    3.392 ERROR     [WARNING]: Invalid characters were found in group names but not replaced, use
    3.392 ERROR    -vvvv to see details
    3.392 ERROR    
    3.392 ERROR    Parsed /var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/plugins/inventory/ec2.py inventory source with script plugin
    3.392 INFO     Processing JSON output...
    3.392 INFO     Loaded 24 groups, 1 hosts
2020-09-08 16:18:43,605 DEBUG    awx.main.models.inventory Going to update inventory computed fields, pk=30
2020-09-08 16:18:43,649 DEBUG    awx.main.models.inventory Finished updating inventory computed fields, pk=30, in 0.041 seconds
    3.732 INFO     Inventory import completed for qa-source in 2.0s `

##### ACTUAL RESULTS

AWS inventory update job and AWS playbook run to fetech AMi and create ec2 error. Where ansible using python 3.6
    `line 705, in get_http_connection\n    return self.new_http_connection(host, port, is_secure)\n  File \"/var/lib/awx/venv/ansible/lib/python3.6/site-packages/boto/connection.py\", line 747, in new_http_connection\n    connection = self.proxy_ssl(host, is_secure and 443 or 80)\n  File \"/var/lib/awx/venv/ansible/lib/python3.6/site-packages/boto/connection.py\", line 796, in proxy_ssl\n    sock.sendall(\"CONNECT %s HTTP/1.0\\r\\n\" % host)\nTypeError: a bytes-like object is required, not 'str'\n" `

  
I have tried to build the images multiple times for but not able to get the right combination as in official v7.0.0 images
To be specific in dir:
/var/lib/awx/venev/awx/lib   -- I see python3.6 and no python2.7 (official task,web image has python2.7 )
/var/lib/awx/venev/awx/lib64   -- I see python3.6 which is ok I guess

/var/lib/awx/venev/ansible/lib   --  I see python3.6 and no python2.7 (official task,web image has python2.7 ) 
/var/lib/awx/venev/ansible/lib64   -- I see python3.6 which is ok I guess

Got no error while building images 

To summarize: 
Get no ansible module installed while running the jobs, had to be manually addressed
Don't get appropriate ansible=2.8.4 + python2.7 interpreter combination which fails all the AWS modules related jobs

Would really appreciate any help. Have followed all the steps as provided in the docs, but issue still persists

**Comment by ryanpetrello** [comment]
Hey @ravigawade,

This issue tracker is for tracking feature enhancements and bugs to AWX itself.  Given its age, it's possible that something about the 7.0.0 image has suffered some bitrot, but I can't say what without some digging.

If you need help troubleshooting an AWX install, you might have better luck on our mailing list or IRC channel; we generally reserve this issue tracker for tracking new bugs and enhancements.

http://webchat.freenode.net/?channels=ansible-awx
https://groups.google.com/forum/#!forum/awx-project
